diff --git a/src/imgui/examples/example_android_opengl3/CMakeLists.txt b/src/imgui/examples/example_android_opengl3/CMakeLists.txt
index 63531f4..c8059c3 100644
--- a/src/imgui/examples/example_android_opengl3/CMakeLists.txt
+++ b/src/imgui/examples/example_android_opengl3/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.10)
 
 project(ImGuiExample)
 
diff --git a/src/imgui/examples/example_glfw_vulkan/CMakeLists.txt b/src/imgui/examples/example_glfw_vulkan/CMakeLists.txt
index 0093ac4..399c8d0 100644
--- a/src/imgui/examples/example_glfw_vulkan/CMakeLists.txt
+++ b/src/imgui/examples/example_glfw_vulkan/CMakeLists.txt
@@ -3,7 +3,7 @@
 #  cd build
 #  cmake -g "Visual Studio 14 2015" ..
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.10)
 project(imgui_example_glfw_vulkan C CXX)
 
 if(NOT CMAKE_BUILD_TYPE)

