From 4975564a1bab501d7ac406648f0df375e5b60279 Mon Sep 17 00:00:00 2001
From: Jan Schultke <me@eisenwave.net>
Date: Sat, 5 Aug 2023 14:20:55 +0200
Subject: [PATCH] add missing include <cstdint>

---
 include/reactphysics3d/configuration.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/reactphysics3d/configuration.h b/include/reactphysics3d/configuration.h
index 6f2de958..6281c93c 100644
--- a/include/reactphysics3d/configuration.h
+++ b/include/reactphysics3d/configuration.h
@@ -27,6 +27,7 @@
 #define	REACTPHYSICS3D_CONFIGURATION_H
 
 // Libraries
+#include <cstdint>
 #include <limits>
 #include <cfloat>
 #include <utility>
