--- a/testbed/CMakeLists.txt
+++ b/testbed/CMakeLists.txt
@@ -10,32 +10,6 @@
 set(NANOGUI_INSTALL       OFF CACHE BOOL " " FORCE)
 set(NANOGUI_BACKEND       OpenGL CACHE BOOL " " FORCE)
 
-# ---- Make sure to recursively clone all the git submodules for external libraries (nanogui) --- #
-find_package(Git QUIET)
-if(GIT_FOUND)
-    if (EXISTS "${CMAKE_SOURCE_DIR}/.git")
-    	# Update submodules as needed
-    	option(CLONE_GIT_SUBMODULES "Check submodules during build" ON)
-    	if(CLONE_GIT_SUBMODULES)
-        	message(STATUS "Git Submodules update")
-		message(STATUS "Working directory ${CMAKE_CURRENT_SOURCE_DIR}")
-        	execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
-                        	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                        	RESULT_VARIABLE GIT_SUBMOD_RESULT)
-        	if(NOT GIT_SUBMOD_RESULT EQUAL "0")
-            	    message(FATAL_ERROR "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
-        	endif()
-    	endif()
-    else()
-  	message(FATAL_ERROR "This is not a Git repository. In order to build the testbed application, you need to clone the ReactPhysics3D repository.")
-    endif()
-else()
-  message(FATAL_ERROR "Git has not been found on your system. This is necessary to build the testbed application because git submodules command is used to get the testbed dependencies (nanogui library). You need to fill in the GIT_EXECUTABLE CMake variable with the path to the Git executable on your system to continue.")
-endif()
-
-# Ask Nanogui not to build shared libraries
-SET(NANOGUI_BUILD_SHARED OFF CACHE BOOL "Build Nanogui with static libraries" FORCE)
-
 # Add the configurations from nanogui
 add_subdirectory(extern/nanogui)
