diff --git a/src/html.c b/src/html.c
index dc64cf4b8..da119c4e9 100644
--- a/src/html.c
+++ b/src/html.c
@@ -135,36 +135,47 @@ gchar *sc_html_parse(SC_HTMLParser *parser)
 static SC_HTMLState sc_html_read_line(SC_HTMLParser *parser)
 {
 	gchar buf[SC_HTMLBUFSIZE];
-	gchar buf2[SC_HTMLBUFSIZE*4];
-	gint index;
-	gint n;
+//	gchar buf2[SC_HTMLBUFSIZE*4];
+	gchar *invalid, *head, *tail;
+	gint index, n;
+	gint tail_size = 0;
 
-	if (parser->fp == NULL)
-		return SC_HTML_EOF;
+	invalid = NULL; tail = NULL;
 
-	n = claws_fread(buf, 1, sizeof(buf) - 1, parser->fp);
-	if (n == 0) {
-		parser->state = SC_HTML_EOF;
-		return SC_HTML_EOF;
-	} else
-		buf[n] = '\0';
-
-	if (conv_convert(parser->conv, buf2, sizeof(buf2), buf) < 0) {
-		index = parser->bufp - parser->buf->str;
+	do {
+		if (parser->fp == NULL)
+			return SC_HTML_EOF; // check parser->state???
 
-		conv_utf8todisp(buf2, sizeof(buf2), buf);
-		g_string_append(parser->buf, buf2);
+		head = buf;
+		if (tail_size > 0) {
+		    head = g_stpcpy(buf,tail);
+		}
 
-		parser->bufp = parser->buf->str + index;
+		n = claws_fread(head, 1, sizeof(buf) - tail_size - 1, parser->fp);
+		if (n == 0) {
+			parser->state = SC_HTML_EOF;
+			g_free(tail);
+			return SC_HTML_EOF;
+		} else {
+			head[n] = '\0';
+			g_free(tail);
+			tail_size = 0;
+        }
+
+
+		if (!g_utf8_validate(buf,-1,&invalid)) {
+			tail_size = head + n - invalid;
+			tail = g_strdup(invalid);
+			buf[n - tail_size] = '\0';
+		}
 
-		return SC_HTML_CONV_FAILED;
-	}
+		index = parser->bufp - parser->buf->str;
 
-	index = parser->bufp - parser->buf->str;
+		g_string_append(parser->buf, buf);
 
-	g_string_append(parser->buf, buf2);
+		parser->bufp = parser->buf->str + index;
 
-	parser->bufp = parser->buf->str + index;
+	} while (tail_size !=0);
 
 	return SC_HTML_NORMAL;
 }
