diff --git a/src/procmime.c b/src/procmime.c
index e9f37529f..4dc51fe56 100644
--- a/src/procmime.c
+++ b/src/procmime.c
@@ -823,7 +823,21 @@ gboolean procmime_scan_text_content(MimeInfo *mimeinfo,
 		ertf_parser_destroy(parser);
 		conv_code_converter_destroy(conv);
 	} else if (mimeinfo->type == MIMETYPE_TEXT && mimeinfo->disposition != DISPOSITIONTYPE_ATTACHMENT) {
-		while (claws_fgets(buf, sizeof(buf), tmpfp) != NULL) {
+		gint tail_size = 0;
+
+		while (claws_fgets(buf + tail_size, sizeof(buf) - tail_size, tmpfp) != NULL) {
+			gint len = strlen(buf + tail_size);
+			gint total_len = tail_size + len;
+			const gchar *invalid;
+
+			buf[total_len] = '\0';
+			tail_size = 0;
+
+			if (!g_utf8_validate(buf, -1, &invalid)) {
+				tail_size = buf + total_len - invalid;
+				buf[total_len - tail_size] = '\0';
+			}
+
 			str = conv_codeset_strdup(buf, src_codeset, CS_UTF_8);
 			if (str) {
 				if ((scan_ret = scan_callback(str, cb_data)) == TRUE) {
@@ -837,6 +851,19 @@ gboolean procmime_scan_text_content(MimeInfo *mimeinfo,
 					break;
 			}
 		}
+
+		/* если после EOF остался хвост — добавляем его */
+		if (tail_size > 0) {
+			buf[tail_size] = '\0';
+			str = conv_codeset_strdup(buf, src_codeset, CS_UTF_8);
+			if (str) {
+				scan_ret = scan_callback(str, cb_data);
+				g_free(str);
+			} else {
+				conv_fail = TRUE;
+				scan_ret = scan_callback(buf, cb_data);
+			}
+		}
 	}
 
 	if (conv_fail)
